app/tests/e2e/test_proof_request_models.py

```diff
@@ -0,0 +1,121 @@
+import pytest
+from aries_cloudcontroller import (
+    AcaPyClient,
+    V10PresentationSendRequestRequest,
+    V20PresRequestByFormat,
+    V20PresSendRequestRequest,
+)
+from fastapi import HTTPException
+
+from app.models.tenants import CreateTenantResponse
+from app.routes.verifier import router as verifier_router
+from app.services.event_handling.sse_listener import SseListener
+from app.tests.util.ecosystem_connections import AcmeAliceConnect
+from shared.util.rich_async_client import RichAsyncClient
+
+VERIFIER_BASE_PATH = verifier_router.prefix
+
+
+@pytest.mark.anyio
+@pytest.mark.xfail(raises=HTTPException, strict=True)
+@pytest.mark.parametrize(
+    "name, version, protocol_version",
+    [
+        ("Proof", None, "v1"),
+        (None, "1.0", "v1"),
+        (None, None, "v1"),
+        ("Proof", None, "v2"),
+        (None, "1.0", "v2"),
+        (None, None, "v2"),
+    ],
+)
+async def test_proof_model_failures(
+    issue_credential_to_alice: list,  # pylint: disable=unused-argument
+    acme_acapy_client: AcaPyClient,
+    acme_and_alice_connection: AcmeAliceConnect,
+    alice_member_client: RichAsyncClient,
+    alice_tenant: CreateTenantResponse,
+    name: str,
+    version: str,
+    protocol_version: str,
+):
+
+    acme_connection_id = acme_and_alice_connection.acme_connection_id
+    alice_listener = SseListener(topic="proofs", wallet_id=alice_tenant.wallet_id)
+
+    if protocol_version == "v1":
+        await acme_acapy_client.present_proof_v1_0.send_request_free(
+            body=V10PresentationSendRequestRequest(
+                auto_remove=False,
+                connection_id=acme_connection_id,
+                comment="Test proof",
+                proof_request={
+                    "name": name,
+                    "version": version,
+                    "requested_attributes": {
+                        "THE_SPEED": {
+                            "name": "speed",
+                            "restrictions": [],
+                        }
+                    },
+                    "requested_predicates": {},
+                },
+                auto_verify=True,
+            )
+        )
```

Comment:
nitpick: we should get in the habit of defining models outside of the method argument. So:
```
request_body=V10PresentationSendRequestRequest(
                auto_remove=False,
                connection_id=acme_connection_id,
                comment="Test proof",
                proof_request={
                    "name": name,
                    "version": version,
                    "requested_attributes": {
                        "THE_SPEED": {
                            "name": "speed",
                            "restrictions": [],
                        }
                    },
                    "requested_predicates": {},
                },
                auto_verify=True,
            )

await acme_acapy_client.present_proof_v1_0.send_request_free(body=request_body)
```

Easier to read, and to debug cases where model creation fails vs method fails.

Same with method in the else condition: